### PR TITLE
Update version number to 1.24 pre-release

### PR DESCRIPTION
**Pull request by bradcray**
This puts us back into the "normal" mode of flagging version numbers as being pre-release.

### CODE REVIEW

**Comment by bradcray** [comment]
Attn: @jhoole

**Comment by bradcray** [comment]
@ben-albrecht / @ankingcodes : Is there something we could do for the pair of `masonInteractive.chpl` tests to avoid a proliferation of files that need to be updated with each release?  Maybe leverage `test/compflags/bradc/printstuff/verion.goodstart` or `test/compflags/bradc/printstuff/versionhelp.sh` in some way?

**Comment by ben-albrecht** [comment]
> Is there something we could do for the pair of `masonInteractive.chpl` tests to avoid a proliferation of files that need to be updated with each release?  Maybe leverage `test/compflags/bradc/printstuff/verion.goodstart` or `test/compflags/bradc/printstuff/versionhelp.sh` in some way?

Yeah, that was a simple oversight. Fixed here: https://github.com/chapel-lang/chapel/pull/16579

I can merge that before you merge this one if you'd prefer.

File: test/compflags/bradc/printstuff/versionhelp.sh
Comment (gbtitus) [inline test/compflags/bradc/printstuff/versionhelp.sh:8]: Could simplify the `cat ... | tr` pipe to just `tr -d \"\\n < $CWD/../../../../compiler/main/BUILD_VERSION`.

**Review by gbtitus — COMMENTED** [review]
I ran the shell command from compflags/bradc/printstuff/versionhelp.sh and never got anything except "pre-release (<SHA>)" from it, despite doing `make compiler` without and with your branch.  Is this expected?

File: test/compflags/bradc/printstuff/versionhelp.sh
Comment (bradcray) [reply test/compflags/bradc/printstuff/versionhelp.sh:8]: I didn't try to be at all creative here, but just restored the lines that were here for the 1.23 pre-release (or at least, that's what I think I did).

**Review by gbtitus — APPROVED** [review]
👍

**Pull request merged**
